### PR TITLE
Travis fix

### PR DESCRIPTION
**Pull request by swaythe**
I've adapted the .travis.yml and install.sh from nilearn for pypreprocess. This includes removing tests for Python 2, doing away with the conda environment and updating dependency versions to the latest ones.

The tests still fail but due to Python errors, same as CircleCI.

@bthirion could you please take a look?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,41 +1,65 @@
...
   allow_failures:
-    # allow_failures seems to be keyed on the python version
-    - python: 2.7
+  # allow_failures keyed to python 3.5 & skipping tests.
+    - python: "3.5"
+      env: DISTRIB="travisci" PYTHON_VERSION="3.5" FLAKE8_VERSION="*" SKIP_TESTS="true"
+    - python: "3.7"
+      env: DISTRIB="travisci" PIP_FLAGS="--pre" COVERAGE="true"
+           NUMPY_VERSION="*" SCIPY_VERSION="*" PANDAS_VERSION="*"
+           SCIKIT_LEARN_VERSION="*" JOBLIB_VERSION="*" LXML_VERSION="*"
```
Comment (bthirion) [inline .travis.yml:18]: I guess that this is copy-pasted from nilearn ?

File: continuous_integration/install.sh [hunk 2]
```diff
@@ -35,7 +35,7 @@ print_conda_requirements() {
     #   - for scikit-learn, SCIKIT_LEARN_VERSION is used
     TO_INSTALL_ALWAYS="pip nose pytest"
     REQUIREMENTS="$TO_INSTALL_ALWAYS"
-    TO_INSTALL_MAYBE="python numpy scipy matplotlib scikit-learn pandas flake8 networkx configobj"
+    TO_INSTALL_MAYBE="numpy scipy matplotlib scikit-learn nilearn nipype pandas flake8 lxml joblib networkx configobj"
```
Comment (bthirion) [inline continuous_integration/install.sh:38]: lxml ?

File: continuous_integration/install.sh [hunk 3]
```diff
@@ -44,70 +44,80 @@ print_conda_requirements() {
...
+#     if [[ "$INSTALL_MKL" == "true" ]]; then
+#         # Make sure that MKL is used
+#         conda install --quiet --yes mkl
+#     elif [[ -z $CIRCLECI ]]; then
+#         # Travis doesn't use MKL but circle ci does for speeding up examples
+#         # generation in the html documentation.
+#         # Make sure that MKL is not used
+#         conda remove --yes --features mkl || echo "MKL not installed"
+#     fi
+# }
```
Comment (bthirion) [inline continuous_integration/install.sh:104]: Why did you add this  ?

**Comment by bthirion** [comment]
travis still failing...

File: .travis.yml (hunk 1 above)
Comment (swaythe) [reply .travis.yml:18]: Yes, it was.

File: continuous_integration/install.sh (hunk 2 above)
Comment (swaythe) [reply continuous_integration/install.sh:38]: Again copied from nilearn. I figured I would remove the extra packages once the required changes are made to pypreprocess and Travis no longer fails.

File: continuous_integration/install.sh (hunk 3 above)
Comment (swaythe) [reply continuous_integration/install.sh:104]: That was part of the original travis.yml file, I forgot to remove it!

**Comment by swaythe** [comment]
> travis still failing...

Yes, but not it is in a different place, corresponding to changes we need to make related to nistats changes.

**Comment by swaythe** [comment]
I will re-issue this PR after trimming the travis.yml and install.sh files.
